QuadradiusR/scripts/game_manager.gd

```diff
@@ -13,16 +13,28 @@ func _ready():
 # # # # # # # # #
 func _game_state_diff(data: Dictionary):
 	var diff = data['game_state_diff']
-	if diff['current_player_id'] == Context.user_id:
-		# Play opponent's move
-		if 'pieces' in diff['board']:
-			for piece_id in diff['board']['pieces'].keys():
-				if piece_id[0] == "$":
-					continue
-				var dest_tile_id = diff['board']['pieces'][piece_id]['tile_id']
-				var source_tile_pos = Context.game_state.get_tile_pos_by_piece_id(piece_id)
-				var dest_tile_pos = Context.game_state.get_tile_pos_by_tile_id(dest_tile_id)
-				board.move_torus_by_tiles(source_tile_pos, dest_tile_pos)
+	if 'pieces' in diff['board']:
+		for piece_id in diff['board']['pieces'].keys():
+			if piece_id[0] == "$":
+				continue
+			
+			var piece = diff['board']['pieces'][piece_id]
+			if board.get_torus_by_id(piece_id).player_id == Context.user_id:
+				continue
+			
+			var source_tile_pos = Context.game_state.get_tile_pos_by_piece_id(piece_id)
+			var dest_tile_pos = Context.game_state.get_tile_pos_by_tile_id(piece['tile_id'])
+			
+			board.move_torus_by_tiles(source_tile_pos, dest_tile_pos)
+	if 'powers' in diff['board']:
+		for orb_id in diff['board']['powers']:
+			var orb = diff['board']['powers'][orb_id]
+			
+			if orb['tile_id']:
+				board.spawn_orb_on_tile_id(orb_id, orb['tile_id'])
+			if orb['piece_id']:
+				board.get_tile_by_orb_id(orb_id).collect_orb()
+				board.get_torus_by_id(orb['piece_id']).add_power(orb_id, Context.game_state.get_powerup_by_id(orb_id)['power_definition_id'])
 	Context.game_state.apply_gamestate_diff(diff)
-	# BUMP ETAGS
+	# TODO BUMP ETAGS
```

Comment:
What is etag?

Reply:
from wiki: The ETag (or entity tag) HTTP response header is an identifier for a specific version of a resource.
Basically it lets us easily track the gamestate versions and fix any inconsistencies